osclib/stagingapi.py

```diff
@@ -1072,25 +1072,27 @@ def map_ring_package_to_subject(self, project, pkg):
 
         return project
 
-    def get_sub_packages(self, pkg, project=None):
+    def get_sub_packages(self, package, project=None):
         """
         Returns a list of packages that need to be linked into rings
         too. A package is actually a tuple of project and package name
         """
         ret = []
         if not project:
-            project = self.ring_packages.get(pkg)
+            project = self.ring_packages.get(package)
         if not project:
             return ret
-        url = self.makeurl(['source', project, pkg],
+        url = self.makeurl(['source', project, package],
                            {'cmd': 'showlinked'})
 
         # showlinked is a POST for rather bizzare reasons
         f = http_POST(url)
         root = ET.parse(f).getroot()
 
         for pkg in root.findall('package'):
-            ret.append((pkg.get('project'), pkg.get('name')))
+            # ensure sub-package is valid
+            if pkg.get('project') in self.rings and pkg.get('name') != package:
```

Comment:
We're misunderstand each other?! Let me ask several question(please don't considering #948 in this change):
1) do you agree `get_sub_packages` is mainly for ring package only? if you don't think so, please point me where is the real case(not returned a empty list https://github.com/openSUSE/osc-plugin-factory/blob/master/osclib/stagingapi.py#L1084) this function is used for adi in the code.
2) do you agree all `valid` sub-package link in Rings must from/to another ring? ie. ring1->ring1, ring1->ring2, etc. is valid; linking to `home:` or whatever else is invalid.